### PR TITLE
chore: Add code generation to Makefile

### PR DESCRIPTION
**Pull request by ckadner**
* Pre-install grpc-gateway plugin in developer Docker image
* Add `generate` make target to run protoc and automatically download required .proto files
* Add developer instructions to README.md

### CODE REVIEW

**Comment by ckadner** [comment]
@rafvasq -- when you get a chance, could you try the revised developer setup and instructions?

`make generate` or `make run generate` should work on a fresh clone and/or after deleting your previously downloaded `google/api/*.proto` files.

**Review by rafvasq — CHANGES_REQUESTED** [review]
Thanks @ckadner. The developer setup/instructions look good aside from the small typo below and both `make generate` and `make run generate` on a fresh clone worked for me.

File: README.md [hunk 1]
```diff
@@ -2,18 +2,50 @@
...
+DOCKER_TAG="dev"
+docker tag kserve/rest-proxy:latest ${DOCKER_USER}/rest-proxy:${DOCKER_TAG}
+docker push ${DOCKER_USER}/rest-proxy:${DOCKER_TAG}
 ```
+
+### Update your ModelMesh deployment
+
+In order to use the newly built `rest-proxy` image in a [Modelmesh Serving deployment](https://github.com/kserve/modelmesh-serving#modelmesh-serving) update the `restProxy.image` in [config/default/config-defaults.yaml](https://github.com/kserve/modelmesh-serving/blob/v0.11.0/config/default/config-defaults.yaml#L31-L32) and (re)deploy the Modelmesh Serving.
+
+To update a running deployment of ModelMesh serving, add or update the `restProxy.image` section in the `model-serving-config` `ConfigMap` as described the [Modelmesh Serving configuration instructions](https://github.com/kserve/modelmesh-serving/tree/main/docs/configuration).
```
[the remark this answers is not in the document]
Comment (ckadner) [reply README.md]: Good nit 🧶

**Review by rafvasq — APPROVED** [review]
/lgtm

**Review by rafvasq — APPROVED** [review]

**Pull request merged**